0.data_analysis/utils/CorrelateData.py

```diff
@@ -0,0 +1,174 @@
+from collections import defaultdict
+from itertools import combinations, product
+
+import pandas as pd
+
+
+class CorrelateData:
+    """
+    Compute inter or intra group correlations between objects.
+    """
+
+    def __init__(self):
+        pass
+
+    def save_params(self, _groups, _cols, _drop_cols, _corrs):
```

Comment:
Recommend type hints in the function. The below documentation is really good! It would just add more documentation if you add type hints.

Reply:
This suggestion applies to all functions in this file.

Reply:
I may do this in the future. Is there a way to provide a type hint if a parameter can be multiple types? Also, do you know of any formmater or tool that will create type hints in python?

Reply:
I do not think there is a formmater for type hints but here is an article about the `typing` module for you to reference on how to add type hint for multiple types. TLDR you can use Union.

https://www.restack.io/docs/hasura-knowledge-hasura-type-hinting-multiple-types